pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -538,6 +600,58 @@ public String convert(Object value, PinotDataType sourceType) {
     }
   },
 
+  JSON {
+    @Override
+    public int toInt(Object value) {
```

Comment:
We should parse the json and convert to value if the json is a value node, or use the same method as the STRING type because value node json can be read as plain string

Reply:
Fixed.